src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/CertificateRequest.cs

```diff
@@ -611,6 +667,12 @@ public X509Certificate2 CreateSelfSigned(DateTimeOffset notBefore, DateTimeOffse
                 {
                     return certificate.CopyWithPrivateKey(mldsa);
                 }
+
+                SlhDsa? slhDsa = _key as SlhDsa;
+                if (slhDsa is not null)
```

Comment:
```suggestion
                SlhDsa? slhDsa = _key as SlhDsa;

                if (slhDsa is not null)
```

Or replace the whole thing with a switch. I don't think it's "wrong" to make code you are touching better.